### DISCUSSION TITLE
Welcome to MonaServer2 Discussions!

### DISCUSSION THREAD

**Discussion by thomasjammet** [MonaSolutions/MonaServer2#43] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xStreamPros** [comment]
Hi Thomas,

I have been working on different scenarios on different installs (Linux and Windows) and have a few questions.

I am wondering if a single server can host several rtmp streams (add additional ports to the .ini file as per the install document) and then use the server port for each one to pull into OBS or the like?

For example lets say the .ini configuration contains 2 rtmp ports as per the install documentation (1935 and 1936) and I have 2 GoPros, each streaming to one port. In OBS I add a source to it using IP and port and have both cameras available as individual sources. 

I can currently use 1935 to stream GoPro to and then use OBS to pull the stream into as a media source using IP and port. But run into issues when trying to add a second camera. This is in Linux and Windows with modified .ini file.

You can contact me if you wish here and discuss in private.
![rtmp](https://user-images.githubusercontent.com/27529593/117590089-d0207500-b0fb-11eb-9e81-234cf0a61a27.PNG)

**Reply by thomasjammet** [reply]
Hi,

There is no need to instantiate multiple RTMP servers, you should simply use different names in your rtmp url to identify different sources. For example :
```
rtmp://localhost/directory/stream1
rtmp://localhost/directory/stream2
```

Or is it something you can't do with your GoPro?

**Reply by xStreamPros** [reply]
Thomas - I will try that. Do you recommend manually creating below "www" the folders? For instance:

rtmp://localhost/GoPro/stream1
rtmp://localhost/GoPro/stream2

Where folder would look like:
www -> GoPro -> stream1
www -> GoPro -> stream2

And then in GoPro cameras use the same rtmp address and port with the source directory for each GoPro?

I have the servers running on Ubuntu Linux amd Windows VM and port forwarding for individual ports to servers.

I may be over complicating it. 

May be best to discuss via email. Plus explore some support for the project.

**Reply by thomasjammet** [reply]
Yes you can send us a mail to discuss further.

Anyway the directory is not really important, it's just that some clients implementations of RTMP needs it. In MonaServer every stream is global, it does not refer to a directory.

So in your case you can just create the directory "GoPro" (or "gopro" to avoid errors of typing). And stream1 or stream2 will be the name of the stream, to identify it. So do not create a directory for each stream name.

Then it should work for you, you can also use MonaServer directly on windows instead of using a VM, it's up to you.

**Comment by Aquilone64** [comment]
Hi, is there any way to implement a low bitrate switcher to change the scene ?

thanks 4 this great project!

**Reply by MathieuPOUX** [reply]
MonaTiny supports MBR but nobody knows it :-)
Just publish multiple bitrate streams with the name that you want, for example: LowStream, NormalStream and HighStream, and play it like this:
`http://localhost/LowStream.mp4?mbr=NormalStream|HighStream`
it will start to play LowStream and can switch on NormalStream or HighStream on congestion.
To start to play NormalStream simply change by:
`http://localhost/NormalStream.mp4?mbr=LowStream|HighStream`
One demonstration here: https://www.youtube.com/watch?v=RXHGZadYU0M
